### ISSUE TITLE
Unhandled exception in thread started by <bound method dbgpSocket._getIncomingDataPacket of <dbgp.client.dbgpSocket instance at 0x10831c5a8>>

### ISSUE THREAD

**Issue by davefultz** [Komodo/KomodoEdit#3434]
### Short Summary

Regularly (but not always) getting this error report at the close of a debugging session:

Unhandled exception in thread started by <bound method dbgpSocket._getIncomingDataPacket of <dbgp.client.dbgpSocket instance at 0x10831c5a8>>
Traceback (most recent call last):
  File "/Applications/Komodo IDE 11.app/Contents/SharedSupport/dbgp/pythonlib/dbgp/client.py", line 2212, in _getIncomingDataPacket
    log.debug("    %d[%r]" , len(data), data)
AttributeError: 'NoneType' object has no attribute 'debug'

### Steps to Reproduce

Run large (50,000 lines) python 2.7.x program in the debugger. It prints 1K – 1M Unicode debugging data to stdout. Every 3 or four runs that exception report is appended to the tail of the run.

### Expected results

No report.

### Actual results

Occasional exceptions (that do not impact the actual run of the code).

### Platform Information

*Komodo Edit or IDE?* 
IDE

*Komodo Version?* 
11.0.2 Build 90813

*Operating System (and version)?* 
macOS 10.12.6

**Comment by mitchell-as** [comment]
Hi, thanks for the report. I presume you do not have a script on hand that I can execute that would trigger this exception? If not, I'm not really sure what to do here. There are plenty of "log.debug()" calls in the same loop as the one triggering the exception, and I see no reason to believe why "log" would be "None" all of a sudden. A quick fix would be to wrap the "log.debug()" call in an "if log: log.debug()" statement, but perhaps others would need to be wrapped too, and it's impossible to say which ones should.

Long story short, if I cannot reliably reproduce this bug, I will not be able to fix it.

**Comment by davefultz** [comment]
Thanks for the note. As I mentioned it's not consistent so I didn't expect a ready answer. I'll keep watching it to see if I can detect an etiology.
